src/mlpack/methods/ann/loss_functions/poisson_nll_loss_impl.hpp

```diff
@@ -45,7 +45,10 @@ PoissonNLLLoss<InputDataType, OutputDataType>::Forward(
   if (logInput)
     loss = arma::exp(input) - target % input;
   else
+  {
+    CheckProbs(input);
```

Comment:
I don't think we need to check it each time. I guess the input could be > 1 and if it actually was < 0, then the result would be NAN.

Reply:
Correct me if I'm wrong. I am going to remove these checks after merge.

Reply:
The changes are not necessary since most of the time it is used without the softmax layer and with `logInput = true`.

Reply:
@lozhnikov Should I revert the changes?

Reply:
I'm not sure, it's up to you. I think both cases are correct.

Reply:
Then I am keeping it this way.